src/core_tests.rs

```diff
@@ -429,37 +429,37 @@ fn sort_nodes_by_distance_to(nodes: &mut [TestNode], name: &XorName) {
     nodes.sort_by(|node0, node1| name.cmp_distance(&node0.name(), &node1.name()));
 }
 
-// Verify that the kademlia invariant is upheld for the node at `index`.
-fn verify_kademlia_invariant_for_node(nodes: &[TestNode], index: usize) {
+/// Verify that the invariant is upheld for the node at `index`.
+fn verify_invariant_for_node(nodes: &[TestNode], index: usize) {
     let routing_tables = nodes.iter().map(|node| node.routing_table()).collect_vec();
-    verify_kademlia_invariant(&routing_tables, index);
+    verify_invariant(&routing_tables, index);
 }
 
-/// Verify that the kademlia invariant is upheld for the routing table at `index`.
-pub fn verify_kademlia_invariant(routing_tables: &[RoutingTable<XorName>], index: usize) {
-    let target = routing_tables[index].our_name();
-    let mut count = routing_tables.len() - 1;
-    let mut bucket_index = 0;
-
-    while count > 0 {
-        let actual_bucket = actual_names_in_bucket(&routing_tables[index], bucket_index);
-        let expected_bucket = expected_names_in_bucket(routing_tables, target, bucket_index);
-        if actual_bucket.len() < GROUP_SIZE {
-            assert!(expected_bucket == actual_bucket,
-                    "Node: {:?}, expected: {:?}. found: {:?}",
-                    target,
-                    expected_bucket,
-                    actual_bucket);
-        }
-        count -= expected_bucket.len();
-        bucket_index += 1;
-    }
+/// Verify that the invariant is upheld for the routing table at `index`.
+pub fn verify_invariant(_routing_tables: &[RoutingTable<XorName>], _index: usize) {
+    // let target = routing_tables[index].our_name();
```

Comment:
Could this not be that our group is => GROUP_SIZE and our neighbor group. I wonder do we need more than that? Be good to understand the invariant here.